hr_payroll_account_operating_unit/models/hr_payslip.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-17 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2016-17 Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo.tools.translate import _
+from odoo import api, models
+from odoo.exceptions import UserError
+
+
+class HrPayslip(models.Model):
+
+    _inherit = 'hr.payslip'
+
+    @api.multi
+    def write(self, vals):
+        res = super(HrPayslip, self).write(vals)
+        if 'move_id' in vals and vals['move_id']:
+            for slip in self:
+                if slip.contract_id and slip.contract_id.operating_unit_id:
+                    slip.move_id.write({'operating_unit_id':
+                                        slip.contract_id.operating_unit_id.id})
+                    if slip.move_id.line_ids:
+                        slip.move_id.line_ids.\
+                            write({'operating_unit_id':
```

Comment:
In v9 it was done by passing the operating unit in the context and then overriding the create method in the account_move. But maybe this is good so leave it like this.